<launch>
    <arg name="use_camera_manager" default="false"/>
    <arg name="camera_name" default="image_color_rect"/>
    <arg name="is_side_cam" default="false"/>

    <arg if="$(arg use_camera_manager)"     name="nodelet_command"  value="load"/>
    <arg if="$(arg use_camera_manager)"     name="manager_name"     value="/camera_nodelet_manager"/>
    <arg unless="$(arg use_camera_manager)" name="nodelet_command"  value="standalone"/>
    <arg unless="$(arg use_camera_manager)" name="manager_name"     value=""/>

    <node pkg="nodelet" type="nodelet" name="cone_projector" output="screen"
          args="$(arg nodelet_command) rr_platform/pointcloud_projector $(arg manager_name)"
          ns="$(arg camera_name)" required="true">
        <param name="camera_info_topic" value="camera_info"/>
        <param name="camera_tf_frame" value="$(arg camera_name)"/>
        <param name="image_topic_in" value="cones/height/debug_bodies_mask"/>
        <param name="pointcloud_topic_out" value="cones/cloud"/>
        <param name="load_timeout" value="300"/>

        <param unless="$(arg is_side_cam)" name="max_forward" value="5.0"/>
        <param unless="$(arg is_side_cam)" name="downsample_factor" type="int" value="2"/>

        <param if="$(arg is_side_cam)" name="max_forward" value="10000.0"/>
        <param if="$(arg is_side_cam)" name="downsample_factor" type="int" value="1"/>
    </node>
</launch>
